### PR TITLE
fix: #1690 unable to identify shebang when it has initial spaces 

### PR DESCRIPTION
**Pull request by yihong0618**
fix: https://github.com/namhyung/uftrace/issues/1690
Signed-off-by: Yi Hong [zouzou0208@gmail.com](mailto:zouzou0208@gmail.com)

### CODE REVIEW

File: utils/utils.h [hunk 1]
```diff
@@ -327,6 +327,31 @@ static inline char *has_kernel_filter(char *buf)
...
+	char *ret;
+
+	if (!string)
+		return NULL;
+	while (*string) {
+		if (!isspace(*string))
+			break;
+		string++;
+	}
+	ret = string;
```
[the remark this answers is not in the document]
Comment (yihong0618) [reply utils/utils.h:342]: copy that, thanks

**Comment by yihong0618** [comment]
@namhyung fixed thank you very much for your kind review

**Comment by honggyukim** [comment]
Hi @yihong0618, thanks for working on this.
```
$ git log -1
From 667ffe14ed554585a613291f2ba9b9de9fc67474 Mon Sep 17 00:00:00 2001
From: yihong0618 <zouzou0208@gmail.com>
Date: Wed, 10 May 2023 22:08:21 +0800
Subject: [PATCH] record: skip whitespaces after shebang for scripts

Python tracing won't work when the shebang line has a space like below:

Fixed: #1690

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
```
Could you please write your real name in your git author name instead of `yihong0618`?

**Comment by yihong0618** [comment]
> yihong0618

sorry I changed it but DCO failed

**Comment by honggyukim** [comment]
> Signed-off-by: yihong <zouzou0208@gmail.com>

Maybe can I ask you to do this way? For example, I assume your name is "Yi Hong". (sorry if not)
```
$ git config --global user.name "Yi Hong"

$ git commit -s --amend --author "Yi Hong <zouzou0208@gmail.com>"
```

**Comment by yihong0618** [comment]
@honggyukim thanks a lot, really learned a lot.

**Comment by honggyukim** [comment]
It looks good now.
```
From 42dc0dd329fb881c884b8ec8f907a5dcc1ca5a6f Mon Sep 17 00:00:00 2001
From: Yi Hong <zouzou0208@gmail.com>
Date: Wed, 10 May 2023 22:08:21 +0800
Subject: [PATCH] record: skip whitespaces after shebang for scripts

Python tracing won't work when the shebang line has a space like below:

Fixed: #1690
Signed-off-by: Yi Hong <zouzou0208@gmail.com>
```
I'm sorry for asking another change again, but your description mentions

> when the shebang line has a space like below:

But it doesn't show the info at the BELOW. Could you rewrite it this way?
```
Subject: [PATCH] record: skip whitespaces after shebang for scripts

Python tracing won't work when the shebang line has a space like below:

  #! /usr/bin/env python3

This patch makes uftrace to understand the above shebang as well.

Fixed: #1690
Signed-off-by: Yi Hong <zouzou0208@gmail.com>
```
If you don't want to update again, then it's okay fine as is.

**Comment by yihong0618** [comment]
> It looks good now.
> 
> ```
> From 42dc0dd329fb881c884b8ec8f907a5dcc1ca5a6f Mon Sep 17 00:00:00 2001
> From: Yi Hong <zouzou0208@gmail.com>
> Date: Wed, 10 May 2023 22:08:21 +0800
> Subject: [PATCH] record: skip whitespaces after shebang for scripts
> 
> Python tracing won't work when the shebang line has a space like below:
> 
> Fixed: #1690
> Signed-off-by: Yi Hong <zouzou0208@gmail.com>
> ```
> 
> I'm sorry for asking another change again, but your description mentions
> 
> > when the shebang line has a space like below:
> 
> But it doesn't show the info at the BELOW. Could you rewrite it this way?
> 
> ```
> Subject: [PATCH] record: skip whitespaces after shebang for scripts
> 
> Python tracing won't work when the shebang line has a space like below:
> 
>   #! /usr/bin/env python3
> 
> This patch makes uftrace to understand the above shebang as well.
> 
> Fixed: #1690
> Signed-off-by: Yi Hong <zouzou0208@gmail.com>
> ```
> 
> If you don't want to update again, then it's okay to skip it.

will do this now.

**Comment by yihong0618** [comment]
@honggyukim Done and thanks again

**Review by honggyukim — APPROVED** [review]
LGTM. Thanks for your work!

**Review by namhyung — APPROVED** [review]
LGTM

**Comment by honggyukim** [comment]
Hmm.. Shouldn't we handle the trailing whitespaces?

We may need another `str_rtrim` or create another wrapper `str_trim` for both left and right trim.

**Comment by GabrielKimm** [comment]
I'll try it.

**Pull request merged**